ibis/backends/flink/registry.py

```diff
@@ -34,6 +34,35 @@ def extract_field_formatter(translator: ExprTranslator, op: ops.Node) -> str:
     return extract_field_formatter
 
 
+def _cast(translator: ExprTranslator, op: ops.generic.Cast) -> str:
+    from ibis.expr.datatypes.core import Timestamp
+
+    arg, to = op.arg, op.to
+    if isinstance(to, Timestamp) and to.timezone:
+        arg_translated = translator.translate(arg)
+        return f"CONVERT_TZ(CAST({arg_translated} AS STRING), 'UTC+0', '{to.timezone}')"
```

Comment:
Doesn't this return a string in the end? Should it return a timestamp?

Reply:
Thanks, great catch! Wrapped it around with `TO_TIMESTAMP()`.